libfwupd/fwupd-common.c

```diff
@@ -348,10 +348,6 @@ fwupd_build_machine_id (const gchar *salt, GError **error)
 		}
 	}
 	if (fn == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_READ,
-				     "The machine-id is not present");
 		return NULL;
```

Comment:
functions returning with failure in Glib always have to set a GError -- you can remove this part of the patch -- see below.